src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,96 @@
+            throw new CanvasOutOfSyncException("NO CANVAS ASSIGNMENT LINKED");
+        }
+    }
+
+    public void validateCanvasAssignment(AttendanceAssignment assignmentConfigurationFromSetup, Long courseId, AttendanceAssignment attendanceAssignment,
+                                         CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) throws IOException, CanvasOutOfSyncException {
+
+        //Look for changes between the course configuration in the form and in the DB
+        if(!isAssignmentConfigurationSaved(assignmentConfigurationFromSetup, attendanceAssignment)) {
+            LOG.info("Configuration form is different than saved assignment configuration for section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId());
+            throw new CanvasOutOfSyncException("Assignment configuration needs to be saved before pushing to Canvas");
+        }
+
+        //Look for the canvas assignment in canvas
+        Optional<Assignment> assignmentOptional;
+        try {
+            assignmentOptional = canvasApiWrapperService.getSingleAssignment(courseId, oauthToken, attendanceAssignment.getCanvasAssignmentId().toString());
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getSectionId(), e);
+            throw new IOException("Could not connect to Canvas to get assignment");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            LOG.error("Assignment not found in Canvas. Empty optional returned from api call.");
+            throw new IOException("Assignment not found in Canvas");
+        }
+
+        //Looks for discrepancy between the assignment in canvas and in database
+        if (assignmentOptional.get().getPointsPossible().doubleValue() != assignmentConfigurationFromSetup.getAssignmentPoints().doubleValue()) {
+            LOG.debug("Discrepancy between Canvas and DB assignment. Point value of Canvas assignment is: " + assignmentOptional.get().getPointsPossible().doubleValue() +
+                      " and point value of Database assignment is :" + assignmentConfigurationFromSetup.getAssignmentPoints().doubleValue());
+            throw new CanvasOutOfSyncException("DISCREPANCY BETWEEN CANVAS AND DATABASE");
+        }
+    }
+
+    /**
+     * Returns true if configuration in the form and in the db is the same
+     */
+    private boolean isAssignmentConfigurationSaved(AttendanceAssignment assignmentConfigurationFromSetup, AttendanceAssignment attendanceAssignmentSaved) {
+        return assignmentConfigurationFromSetup.getAssignmentPoints().doubleValue() == attendanceAssignmentSaved.getAssignmentPoints().doubleValue()
+                && assignmentConfigurationFromSetup.getExcusedPoints().doubleValue() == attendanceAssignmentSaved.getExcusedPoints().doubleValue()
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1244)